src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -63,7 +63,8 @@ public DistributedApplicationBuilder(DistributedApplicationOptions options)
         _innerBuilder.Services.AddHostedService<DcpHostService>();
 
         // Dashboard
-        _innerBuilder.Services.AddHostedService<DashboardServiceHost>();
+        _innerBuilder.Services.AddSingleton<DashboardServiceHost>();
+        _innerBuilder.Services.AddHostedService<DashboardServiceHost>(sp => sp.GetRequiredService<DashboardServiceHost>());
```

Comment:
Why this change?

Reply:
I use it to grab the dashboard service URI (gRPC endpoint) from the DashboardServiceHost. My long standing gripe about AddHostedService is that it also doesn't inject it as a singleton service :) This is my workaround!